src/components/TokenHistory.js

```diff
@@ -43,7 +50,9 @@ class TokenHistory extends React.Component {
   }
 
   componentDidUpdate = (prevProps) => {
-    if (this.props.history !== prevProps.history) {
```

Comment:
This is a token history screen, not a tx screen. And this `if` is because re-render was not being called when a new tx arrived in the redux.

```
const mapStateToProps = (state) => {
  return { 
    tokensHistory: state.tokensHistory,
  };
};
```

The redux array that must be checked depend on the component props, otherwise I would be able to do something like this

```
const mapStateToProps = (state) => {
  return { 
    tokensHistory: state.tokensHistory[this.props.tokenUID],
  };
};
```

But I don't have access to `this.props`. Is there a way to do it?

Reply:
I guess you have access to `props`. Check it out here: https://react-redux.js.org/using-react-redux/connect-mapstate

Reply:
The refactor was done and the issue was created: https://github.com/HathorNetwork/hathor-wallet/issues/196